pkg/metricsadapter/controller.go

```diff
@@ -73,6 +79,66 @@ func NewMetricsController(restConfig *rest.Config, factory informerfactory.Share
 	return controller
 }
 
+func nodeTransformFunc(obj interface{}) (interface{}, error) {
+	var node *corev1.Node
+	switch t := obj.(type) {
+	case *corev1.Node:
+		node = t
+	case cache.DeletedFinalStateUnknown:
+		var ok bool
+		node, ok = t.Obj.(*corev1.Node)
+		if !ok {
+			return obj, fmt.Errorf("expect resource Node but got %v", reflect.TypeOf(t.Obj))
+		}
+	default:
+		return obj, fmt.Errorf("expect resource Node but got %v", reflect.TypeOf(obj))
+	}
+
+	aggregatedNode := &corev1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              node.Name,
+			Namespace:         node.Namespace,
+			Labels:            node.Labels,
+			DeletionTimestamp: node.DeletionTimestamp,
+		},
+	}
+	return aggregatedNode, nil
+}
+
+func podTransformFunc(obj interface{}) (interface{}, error) {
+	var pod *corev1.Pod
+	switch t := obj.(type) {
+	case *corev1.Pod:
+		pod = t
+	case cache.DeletedFinalStateUnknown:
+		var ok bool
+		pod, ok = t.Obj.(*corev1.Pod)
+		if !ok {
+			return obj, fmt.Errorf("expect resource Pod but got %v", reflect.TypeOf(t.Obj))
+		}
+	default:
+		return obj, fmt.Errorf("expect resource Pod but got %v", reflect.TypeOf(obj))
+	}
+
+	aggregatedPod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              pod.Name,
+			Namespace:         pod.Namespace,
+			Labels:            pod.Labels,
+			DeletionTimestamp: pod.DeletionTimestamp,
+		},
+	}
+	return aggregatedPod, nil
+}
+
+func newInstance() typedmanager.MultiClusterInformerManager {
+	transforms := map[schema.GroupVersionResource]cache.TransformFunc{
+		provider.NodesGVR: cache.TransformFunc(nodeTransformFunc),
+		provider.PodsGVR:  cache.TransformFunc(podTransformFunc),
+	}
+	return typedmanager.NewMultiClusterInformerManager(context.TODO().Done(), transforms)
```

Comment:
Is it possible to use method `GetInstance` in util package and pass `transforms` as an input parameter?

https://github.com/karmada-io/karmada/blob/456d099404d09ba4094614a6a3a8bea36cdf6d61/pkg/util/fedinformer/typedmanager/multi-cluster-manager.go#L42-L51

Reply:
I don't think this is better.
 It would make `GetInstance` more complicated

Reply:
It does add a lot of changes, but what I'm trying to say is that the specific decision of which fields to store should be left up to the caller, as it's currently written in this pr, rather than putting `tarnsforms` in the `util` package.

But that's okay, I agree with the current code.

LGTM